### DISCUSSION TITLE
"Issue with Building stella_vslam on WSL2 Ubuntu 22.04"

### DISCUSSION THREAD

**Discussion by dokodoko123** [stella-cv/stella_vslam#594] [Q&A] [answered]
I am trying to build stella_vslam on WSL2 Ubuntu 22.04 following the instructions at this URL: https://stella-cv.readthedocs.io/en/latest/

An error occurred when I ran the commands listed in the build instructions:
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo ..

The error message is as follows:

CMake Error at src/stella_vslam/CMakeLists.txt:74 (add_library):
  Target "stella_vslam" links to target "g2o::solver_csparse" but the target
  was not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?

CMake Error at src/stella_vslam/CMakeLists.txt:74 (add_library):
  Target "stella_vslam" links to target "g2o::csparse_extension" but the
  target was not found.  Perhaps a find_package() call is missing for an
  IMPORTED target, or an ALIAS target is missing?

I plan to use SocketViewer as the Viewer.

Could you please help me resolve the issue mentioned above? Additionally, if anyone has successfully built stella_vslam on WSL2 Ubuntu, could you share the steps you followed?

**Comment by ymd-stella** [comment, marked as answer]
Have you installed the recommended version of g2o?

**Reply by dokodoko123** [reply]
You were right, the issue was caused by the version of g2o. I had mistakenly built a different version. Thank you for your response.
